driver-core/src/main/com/mongodb/connection/netty/NettyStream.java

```diff
@@ -211,18 +269,32 @@ public void operationComplete(final ChannelFuture future) throws Exception {
 
     @Override
     public void readAsync(final int numBytes, final AsyncCompletionHandler<ByteBuf> handler) {
-        readAsync(numBytes, handler, 0);
+        readAsync(numBytes, handler, readTimeoutMillis);
     }
 
-    private void readAsync(final int numBytes, final AsyncCompletionHandler<ByteBuf> handler, final int additionalTimeout) {
-        scheduleReadTimeout(additionalTimeout);
+    /**
+     * @param numBytes Must be equal to {@link #pendingReader}{@code .numBytes} when called by a Netty channel handler.
+     * @param handler Must be equal to {@link #pendingReader}{@code .handler} when called by a Netty channel handler.
+     * @param readTimeoutMillis Must be equal to {@link #NO_SCHEDULE_TIMEOUT} when called by a Netty channel handler.
+     *                          Timeouts may be scheduled only by the public read methods. Taking into account that concurrent pending
+     *                          readers are not allowed, there must not be a situation when threads attempt to schedule a timeout
+     *                          before the previous one is either cancelled or completed.
+     */
+    private void readAsync(final int numBytes, final AsyncCompletionHandler<ByteBuf> handler, final long readTimeoutMillis) {
         ByteBuf buffer = null;
         Throwable exceptionResult = null;
         synchronized (this) {
             exceptionResult = pendingException;
+            PendingReader pendingReader = this.pendingReader;
```

Comment:
Nit: Does this need to be made a local variable? If so seems we've used `localPendingReader` elsewhere to make it clearer.

Reply:
The reason I am using local variables for shared state is to reduce the number of read usages we can see in the code (including the "Find usages" search in an IDE). The fewer code pieces include reads from or writes to a shared variable, the easier it is (at least to me) to think about the correctness of concurrent code.

The reason I did not call it `localPendingReader` is because `local` adds verbosity without adding any information: we can immediately see that this is a local variable on the line it is declared, and on the other lines where it is used IntelliJ IDEA IDE differentiates local and object variables (by default by using different boldness of the font).

Reply:
There is a cognitive load and cost to a code base by reusing the same named variable. It also adds future risk to the codebase as  the localization of the variable may be changed.

Adding `local` does add verbosity but I disagree because it does add extra information, its explicit. Explicit should be preferred over implicit. Especially as you can't guarantee users have the same IDE or even settings in their IDE as you. 

So please make it explicit, it costs nothing and improves the readability of the code for future development.

Reply:
Our practice generally in this code base is to minimize use of local variables.  I can see the argument here in favor, since this is state that is protected by a lock, but at the same time I do find the style odd, particularly likes 288-290 below.

Reply:
Removed the `pendingReader` local variable.